File: app/src/main/java/com/kota/Bahamut/Pages/BoardPage/BoardSearchPage.java
package com.kota.Bahamut.Pages.BoardPage;

import static com.kota.Bahamut.Service.CommonFunctions.getContextString;

import android.view.View;
import com.kota.ASFramework.Dialog.ASAlertDialog;
import com.kota.Bahamut.BahamutPage;
import com.kota.Bahamut.DataModels.Bookmark;
import com.kota.Bahamut.DataModels.BookmarkStore;
import com.kota.Bahamut.PageContainer;
import com.kota.Bahamut.Pages.Model.BoardPageItem;
import com.kota.Bahamut.R;
import com.kota.Bahamut.Service.TempSettings;
import com.kota.Telnet.TelnetClient;
import com.kota.TelnetUI.TelnetHeaderItemView;

public class BoardSearchPage extends BoardMainPage {
    private String _author = null;
    private String _gy = null;
    private String _keyword = null;
    private String _mark = null;

    public int getPageType() {
        return BahamutPage.BAHAMUT_BOARD_SEARCH;
    }

    public int getPageLayout() {
        return R.layout.board_search_page;
    }

    public synchronized void onPageRefresh() {
        super.onPageRefresh();
        TelnetHeaderItemView header_view = (TelnetHeaderItemView) findViewById(R.id.BoardPage_HeaderView);
        if (header_view != null) {
            String board_name = getListName();
            if (board_name == null) {
                board_name = getContextString(R.string.loading);
            }
            header_view.setData(this._board_title, "文章搜尋", board_name);
        }
    }

    protected boolean onMenuButtonClicked() {
        showSelectArticleDialog();
        return true;
    }

    protected boolean onListViewItemLongClicked(View itemView, int selectedIndex) {
        BoardPageItem item = (BoardPageItem) BoardSearchPage.this.getItem(selectedIndex);

        if (item!=null) {
            ASAlertDialog.createDialog()
                    .setTitle(getContextString(R.string.insert) + getContextString(R.string.bookmark))
                    .setMessage(getContextString(R.string.insert_this_bookmark) + "\n\"" + item.Title + "\"")
                    .addButton(getContextString(R.string.cancel))
                    .addButton(getContextString(R.string.insert))
                    .setListener((aDialog, index) -> {
                        if (index == 1) {
                            Bookmark bookmark = new Bookmark();
                            System.out.println("add bookmark:" + bookmark.getTitle());
                            bookmark.setBoard(BoardSearchPage.this.getListName());
                            bookmark.setKeyword(item.Title);
                            bookmark.setTitle(bookmark.generateTitle());
                            BookmarkStore store = TempSettings.getBookmarkStore();
                            store.getBookmarkList(BoardSearchPage.this.getListName()).addBookmark(bookmark);
                            store.store();
                        }
                    }).scheduleDismissOnPageDisappear(this).show();
        }
        return true;
    }

    public int getListType() {
        return 2;
    }

    public String getListIdFromListName(String aName) {
        return aName + "[Board][Search]";
    }

    protected void onPostButtonClicked() {
        ASAlertDialog.createDialog()
                .setTitle(getContextString(R.string.insert) + getContextString(R.string.bookmark))
                .setMessage(getContextString(R.string.insert_this_bookmark_search))
                .addButton(getContextString(R.string.cancel))
                .addButton(getContextString(R.string.insert))
                .setListener((aDialog, index) -> {
            if (index == 1) {
                Bookmark bookmark = new Bookmark();
                System.out.println("add bookmark:" + bookmark.getTitle());
                bookmark.setBoard(BoardSearchPage.this.getListName());
                bookmark.setKeyword(BoardSearchPage.this._keyword);
                bookmark.setAuthor(BoardSearchPage.this._author);
                bookmark.setMark(BoardSearchPage.this._mark);
                bookmark.setGy(BoardSearchPage.this._gy);
                bookmark.setTitle(bookmark.generateTitle());
                BookmarkStore store = TempSettings.getBookmarkStore();
                store.getBookmarkList(BoardSearchPage.this.getListName()).addBookmark(bookmark);
                store.store();
            }
        }).scheduleDismissOnPageDisappear(this).show();
    }

    public void setKeyword(String keyword) {
        this._keyword = keyword;
    }

    public void setAuthor(String author) {
        this._author = author;
    }

    public void setMark(String mark) {
        this._mark = mark;
    }

    public void setGy(String gy) {
        this._gy = gy;
    }

    protected boolean onBackPressed() {
        clear();
        getNavigationController().popViewController();
        TelnetClient.getClient().sendKeyboardInputToServerInBackground(256, 1);
        PageContainer.getInstance().cleanBoardSearchPage();
        return true;
    }

    public boolean isAutoLoadEnable() {
        return false;
    }
}

File: app/src/main/java/com/kota/Bahamut/Pages/BoardPage/BoardSearchPage.kt
package com.kota.Bahamut.Pages.BoardPage;

import com.kota.Bahamut.Service.CommonFunctions.getContextString

import android.view.View
import com.kota.ASFramework.Dialog.ASAlertDialog
import com.kota.Bahamut.BahamutPage
import com.kota.Bahamut.DataModels.Bookmark
import com.kota.Bahamut.DataModels.BookmarkStore
import com.kota.Bahamut.PageContainer
import com.kota.Bahamut.Pages.Model.BoardPageItem
import com.kota.Bahamut.R
import com.kota.Bahamut.Service.TempSettings
import com.kota.Telnet.Reference.TelnetKeyboard
import com.kota.Telnet.TelnetClient

class BoardSearchPage : BoardMainPage()() {
    private var _author: String = null;
    private var _gy: String = null;
    private var _keyword: String = null;
    private var _mark: String = null;

    getPageType(): Int {
        return BahamutPage.BAHAMUT_BOARD_SEARCH;
    }

    getPageLayout(): Int {
        return R.layout.board_search_page;
    }

    public synchronized Unit onPageRefresh() {
        super.onPageRefresh();
        var header_view: BoardHeaderView = findViewById<BoardHeaderView>(R.id.BoardPage_HeaderView);
        if var !: (header_view = null) {
            var board_name: String = getListName();
            var (board_name: if == null) {
                board_name = getContextString(R.String.loading);
            }
            header_view.setData(_board_title, "文章搜尋", board_name);
        }
    }

    protected fun onMenuButtonClicked(): Boolean {
        showSelectArticleDialog();
        var true: return
    }

    protected fun onListViewItemLongClicked(View itemView, Int selectedIndex): Boolean {
        var item: BoardPageItem = (BoardPageItem) BoardSearchPage.getItem(selectedIndex);

        var (item!: if =null) {
            ASAlertDialog.createDialog()
                    .setTitle(getContextString(R.String.insert) + getContextString(R.String.bookmark))
                    .setMessage(getContextString(R.String.insert_this_bookmark) + "\n\"" + item.Title + "\"")
                    .addButton(getContextString(R.String.cancel))
                    .addButton(getContextString(R.String.insert))
                    .setListener((aDialog, index) -> {
                        var (index: if == 1) {
                            var bookmark: Bookmark = Bookmark();
                            System.out.println("add bookmark:" + bookmark.getTitle());
                            bookmark.setBoard(BoardSearchPage.getListName());
                            bookmark.setKeyword(item.Title);
                            bookmark.setTitle(bookmark.generateTitle());
                            var store: BookmarkStore = TempSettings.bookmarkStore;
                            var (store!: if =null) {
                                store.getBookmarkList(BoardSearchPage.getListName()).addBookmark(bookmark);
                                store.store();
                            }
                        }
                    }).scheduleDismissOnPageDisappear(this).show();
        }
        var true: return
    }

    @Override
    getListType(): Int {
        return BoardPageAction.SEARCH
    }

    @Override
    getListIdFromListName(String aName): String {
        return aName + "[Board][Search]";
    }

    protected fun onPostButtonClicked(): Unit {
        ASAlertDialog.createDialog()
                .setTitle(getContextString(R.String.insert) + getContextString(R.String.bookmark))
                .setMessage(getContextString(R.String.insert_this_bookmark_search))
                .addButton(getContextString(R.String.cancel))
                .addButton(getContextString(R.String.insert))
                .setListener((aDialog, index) -> {
            var (index: if == 1) {
                var bookmark: Bookmark = Bookmark();
                System.out.println("add bookmark:" + bookmark.getTitle());
                bookmark.setBoard(BoardSearchPage.getListName());
                bookmark.setKeyword(BoardSearchPage._keyword);
                bookmark.setAuthor(BoardSearchPage._author);
                bookmark.setMark(BoardSearchPage._mark);
                bookmark.setGy(BoardSearchPage._gy);
                bookmark.setTitle(bookmark.generateTitle());
                var store: BookmarkStore = TempSettings.bookmarkStore;
                var (store!: if =null) {
                    store.getBookmarkList(BoardSearchPage.getListName()).addBookmark(bookmark);
                    store.store();
                }
            }
        }).scheduleDismissOnPageDisappear(this).show();
    }

    setKeyword(String keyword): Unit {
        _keyword = keyword;
    }

    setAuthor(String author): Unit {
        _author = author;
    }

    setMark(String mark): Unit {
        _mark = mark;
    }

    setGy(String gy): Unit {
        _gy = gy;
    }

    protected fun onBackPressed(): Boolean {
        clear();
        getNavigationController().popViewController();
        TelnetClient.getClient().sendKeyboardInputToServerInBackground(TelnetKeyboard.LEFT_ARROW, 1);
        PageContainer.getInstance().cleanBoardSearchPage();
        var true: return
    }

    isAutoLoadEnable(): Boolean {
        var false: return
    }
}


